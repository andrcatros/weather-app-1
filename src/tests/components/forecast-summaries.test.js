import React from 'react'
import { render } from '@testing-library/react'
import ForecastSummaries from '../../components/forecastsummaries'

const mockFunction = () => {
    return
}

describe('Forecast Summaries', () => {
    const forecasts = [
        {
            date: 1525046400000,
            temperature: {
                max: 11
            },
            description: "Clear",
            icon: "800"
        },
        {
            date: 1525046400022,
            temperature: {
                max: 25
            },
            description: "Foggy",
            icon: "900"
        },
    ];

    it("renders the correct amount of ForecastSummary components", () => {
        const { asFragment } = render(
            <ForecastSummaries 
                forecasts={ forecasts } 
                onForecastSelect={ mockFunction }
            />
        );
    
        expect(asFragment()).toMatchSnapshot()
      });

    it('renders a number of props into a weather summary', () => {
        const { getAllByTestId } = render(
            <ForecastSummaries
                forecasts={ forecasts }
                onForecastSelect={ mockFunction }
            />
        )

        expect(getAllByTestId('date-id')).toHaveLength(2);
        expect(getAllByTestId('temperature-id')).toHaveLength(2);
        expect(getAllByTestId('description-id')).toHaveLength(2);
        expect(getAllByTestId('icon-image-id')).toHaveLength(2);
    })
})
        
