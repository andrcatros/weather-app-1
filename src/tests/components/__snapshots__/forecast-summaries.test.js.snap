// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Summaries renders the correct amount of ForecastSummary components 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
    >
      <span
        class="date"
        data-testid="date-id"
      >
        1525046400000
      </span>
      <span
        class="icon"
        data-testid="icon-id"
      >
        800
      </span>
      <span
        class="max-temperature"
        data-testid="temperature-id"
      >
        11
      </span>
      <span
        class="description"
        data-testid="description-id"
      >
        Clear
      </span>
    </div>
    <div
      class="forecast-summary"
    >
      <span
        class="date"
        data-testid="date-id"
      >
        1525046400022
      </span>
      <span
        class="icon"
        data-testid="icon-id"
      >
        900
      </span>
      <span
        class="max-temperature"
        data-testid="temperature-id"
      >
        25
      </span>
      <span
        class="description"
        data-testid="description-id"
      >
        Foggy
      </span>
    </div>
  </div>
</DocumentFragment>
`;
